### ISSUE TITLE
Modern kernels

### ISSUE THREAD

**Issue by roshanr95** [aws/aws-nitro-enclaves-sdk-bootstrap#26]
Are there plans to provide images based on modern kernels? E.g. 6.1 (or 5.15) Amazon Linux mainlines, there quite a lot of useful features in the newer kernels.

**Comment by Cottand** [comment]
Hi @meerd - I am trying to boot a Nitro enclave with a 6.8 kernel blob (and its kernel config). The enclave hangs at boot, suggesting it does not even get to initialising the console. Is there anything in Amazon Linux kernels (as opposed to mainline) that is specific to the enclave and is required to boot?

**Comment by Cottand** [comment]
Thanks @roshanr95 , that helped!

For reference, we have open-sourced a Nitro CLI alternative that allows using modern kernels without having to wait for this repo to provide a new image or a new init binary

see https://github.com/monzo/aws-nitro-util . We have successfully booted a 6.8 Kernel compiled from source in an enclave.

**Comment by roshanr95** [comment]
ayo @Cottand, that repo's amazing :heart_eyes:

**Comment by foersleo** [comment]
We have restructured how we build the binary blobs and are now also offering kernels based on upstream LTS v6.6 series. We plan to improve on the regularity of our kernel updates.

What is still missing at the moment is a new release of the aws-nitro-enclaves-cli to package these newly build binaries.

**Comment by roshanr95** [comment]
Trying it out, nix support is great!

**Comment by roshanr95** [comment]
@foersleo does it need a different cmdline as well? compared to the existing nitro-cli?

**Comment by foersleo** [comment]
Hi @roshanr95, I have run the newer kernels without changes to the cmdline. So, I do not think there is a vital part missing.

Do you run into any problems with the newer kernels with the existing cmdline files?

**Comment by roshanr95** [comment]
Yes, was facing issues, but it turned out to not be the kernel. It was linuxkit actually, I guess nitro-cli needs to be updated to work with newer versions.

**Comment by foersleo** [comment]
Yes, you are right. Through all of this I have not had too close of an eye on getting all of this out into a cli release.

We have had updated the linuxkit binary shipped with the CLI in the source tree a while ago (https://github.com/aws/aws-nitro-enclaves-cli/commit/04f48e0da1fed413545e39b9574c8b0d466457ae) with the necessary adjustments to work with the new linuxkit.

The new CLI with that linuxkit included was just released today as [v1.3.2](https://github.com/aws/aws-nitro-enclaves-cli/releases/tag/v1.3.2), although it will be a bit longer until it is available as an rpm through the Amazon Linux package repositories.

A release of the CLI including the new kernel binaries is still pending some additional testing and preparation, and unfortunately I can not share a firm timeline for that yet.

Sorry, for the confusion.

So, to summarize:

* You can use the latest release 1.3.2 of aws-nitro-enclaves-cli today if you are fine building it from source.
* That release does include the new linuxkit binary alongside the adjustments needed to make the new linuxkit work.
* That release does **not** include the new binaries of `kernel`, `nsm.ko` (Neither `init`, however the only change not included is it being build reproducibly)

I hope this helps. Let me know how we can assist more.

**Comment by roshanr95** [comment]
Perfect, thanks!
